src/app/features/tech-record/create-batch-trl/components/batch-trl-details/batch-trl-details.component.html

```diff
@@ -0,0 +1,33 @@
+<h1 class="govuk-heading-l title">Add batch of trailers</h1>
+<form [formGroup]="form">
+  <app-text-input formControlName="applicationId" name="applicationId" label="Application ID" [width]="width.XL"></app-text-input>
+
+  <div class="govuk-inset-text">
+    Add the VINs for each new trailer. Each trailer will be given a unique Trailer ID <strong>if the Trailer ID field is left blank</strong>.
+  </div>
+
+  <ng-container formArrayName="vehicles">
+    <div class="flex flex--wrap flex_gap--7 justify--flex-start">
+      <ng-container *ngFor="let group of vehicles.controls; let i = index" [formGroupName]="i">
+        <fieldset id="batch-trl-details" class="govuk-fieldset">
+          <legend class="govuk-fieldset__legend govuk-fieldset__legend--m">
+            <h2 class="govuk-fieldset__heading">{{ 'Vehicle ' + (i + 1) }}</h2>
+          </legend>
+          <app-text-input formControlName="vin" name="input-vin{{ i }}" label="VIN" [width]="width.L">
+            <ng-container *ngIf="getVin(group)?.dirty && (getVin(group)?.status === 'VALID' || getVin(group)?.status === 'PENDING')">
+              <ng-template appSuffix>
+                <app-input-spinner [isValid]="getVin(group)?.status || 'PENDING'"></app-input-spinner>
+              </ng-template>
+            </ng-container>
+          </app-text-input>
+          <app-text-input formControlName="trailerId" name="trailerId{{ i }}" label="Trailer ID" [width]="width.L"></app-text-input>
+        </fieldset>
+      </ng-container>
+    </div>
+  </ng-container>
```

Comment:
You can get rid of 2 layers of nesting here.
```suggestion
  <div class="flex flex--wrap flex_gap--7 justify--flex-start" formArrayName="vehicles">
    <fieldset *ngFor="let group of vehicles.controls; let i = index" [formGroupName]="i" id="batch-trl-details" class="govuk-fieldset">
      <legend class="govuk-fieldset__legend govuk-fieldset__legend--m">
        <h2 class="govuk-fieldset__heading">{{ 'Vehicle ' + (i + 1) }}</h2>
      </legend>

      <app-text-input formControlName="vin" name="input-vin{{ i }}" label="VIN" [width]="width.L">
        <ng-container *ngIf="getVin(group)?.dirty && (getVin(group)?.status === 'VALID' || getVin(group)?.status === 'PENDING')">
          <ng-template appSuffix>
            <app-input-spinner [isValid]="getVin(group)?.status || 'PENDING'"></app-input-spinner>
          </ng-template>
        </ng-container>
      </app-text-input>
      <app-text-input formControlName="trailerId" name="trailerId{{ i }}" label="Trailer ID" [width]="width.L"></app-text-input>
    </fieldset>
  </div>
```

Reply:
Addressed this myself.